### DISCUSSION TITLE
Error defining an input parameter in a mutation with the name "input"

### DISCUSSION THREAD

**Discussion by jmbarroso** [Netflix/dgs-framework#187] [Q&A] [answered]
The following schema generates the compilation error: `'input' hides member of supertype 'GraphQLQuery' and needs 'override' modifier` on the generated class **CreateUserGraphQLQuery**

```
type Mutation {
    createUser(input: UserInput): User
}

type User {
    username: String!
}

input UserInput {
    username: String
}
```
 
Is this expected? (problem goes away if the mutation is changed to something like  _createUser(userInput: UserInput): User_

**Comment by srinivasankavitha** [comment, marked as answer]
Thanks for reporting. This is a known issue captured here: https://github.com/Netflix/dgs-codegen/issues/52
